### PR TITLE
Wait for MOTD before joining channels in irc bot

### PR DESCRIPTION
**Pull request by tmarki**
Hello,

your version does not work for the MOTD message, so on some IRC servers it fails to join any channels.

I fixed this by adding a callback for this, I hope you can merge it.

Thanks,
Tamas

### CODE REVIEW

File: discirc/irc_bot.py [hunk 1]
```diff
@@ -70,6 +72,8 @@ def on_connect(self):
         self.send('USER', user=self.nick, realname='Discord gateway')
         if self.password:
             self.send('PASS', password=self.password)
+
+    def on_motddone(self, message):
```
Comment (j0ack) [inline discirc/irc_bot.py:76]: Could you just add a small docstring to the function ?
People may not understand what `motdone` is.

**Review by j0ack — CHANGES_REQUESTED** [review]
Thanks for your work, just a little comment

**Comment by tmarki** [comment]
Added the docstring, I hope it's descriptive enough!

**Comment by TeoTwawki** [comment]
@tmarki if I may make a suggestion:
```python
    """checking if Message Of The Day is done because it may interfere with joining channels"""
    def on_motddone(self, message):
```
Future editors then know both why clients should wait  and what motd is when they read the comment.

j0ack was concerned ppl wouldn't know what motd is.

**Comment by j0ack** [comment]
I agree with @TeoTwawki, people may not know what `MOTD` stands for.

Just a little description as suggested would be great.

**Comment by tmarki** [comment]
I replaced the docstring with your version.

**Comment by j0ack** [comment]
Put the docstring behind the `def` and it will be fine for me.

Thanks for the work :)

**Comment by tmarki** [comment]
Done!

**Pull request merged**